File: Knapsack/Lab01/Result.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab01
{
    internal class Result
    {

        public int totalValue;
        public int totalWeight;

        List<Item> ListOfResults = new List<Item>();

        public Result(List<Item> List) {

            ListOfResults = List;
        }

        public void PrintKnapsackContents() // ToString
        {
            ListOfResults.Sort((x, y) => (y.value * 1.0 / y.weight).CompareTo(x.value * 1.0 / x.weight));

            Console.WriteLine("List of items inside the knapsack:");
            foreach (var item in ListOfResults)
            {
                Console.WriteLine($"Nr. {item.index}        Value: {item.value}      Weight: {item.weight}");
                totalValue += item.value;
                totalWeight += item.weight;
            }
            Console.WriteLine($"Knapsack value: {totalValue}");
            Console.WriteLine($"Knapsack weight: {totalWeight}");

        }
    }
}

File: JAVA/KnapsackProblem/src/main/java/Knapsack/Result.java
package Knapsack;

import java.util.List;

public class Result {
    private List<Item> items;
    private int totalWeight;
    private int totalValue;

    public Result(List<Item> items, int totalWeight, int totalValue) {
        this.items = items;
        this.totalWeight = totalWeight;
        this.totalValue = totalValue;
    }

    public List<Item> getItems() {
        return items;
    }

    public int getTotalWeight() {
        return totalWeight;
    }

    public int getTotalValue() {
        return totalValue;
    }

    public int getItemCount() {
        return items.size();
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Items in knapsack:\n");
        for (Item item : items) {
            sb.append(item.toString()).append("\n");
        }
        sb.append("Total weight: ").append(totalWeight).append("\n");
        sb.append("Total value: ").append(totalValue).append("\n");
        sb.append("Count of items: ").append(getItemCount()).append("\n");
        return sb.toString();
    }
}